### PR TITLE
rethinkdb: 2.3.6 -> 2.4.0, fix

### PR DESCRIPTION
**Pull request by matthias-t**
###### Motivation for this change

The `rethinkdb` package for the RethinkDB database server fails to build and is marked as broken. The `python2.pkgs.rethinkdb` package for the python client errors because of a missing dependency. This fixes the two packages. Please see the commit messages for details.

Also, I changed the license meta fields as the codebase was relicensed to Apache 2.0 after ownership was transferred to the Linux foundation.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/rethinkdb/default.nix [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "e1c1f8ad93bc1c6f2aaa73afc333c57d505d8cc08c437d78a5c1eb8dc4b7e8c2";
+    sha256 = "0x4xg2252k612g9i4n64rimvjy2znc78gmq73v8gbwwa94mjbd2f";
   };
 
   doCheck = false;
 
+  buildInputs = with python2.pkgs; [ six ];
```
Comment (jonringer) [inline pkgs/development/python-modules/rethinkdb/default.nix:18]: in python-modules, the python packages are parameterized per python interpreter, don't bring in interpreter, rather individual packages
```suggestion
  buildInputs = [ six ];
```

File: pkgs/development/python-modules/rethinkdb/default.nix [hunk 2]
```diff
@@ -1,23 +1,26 @@
 { stdenv
 , buildPythonPackage
 , fetchPypi
+, python2
```
Comment (jonringer) [inline pkgs/development/python-modules/rethinkdb/default.nix:4]:
```suggestion
, six
```

File: pkgs/servers/nosql/rethinkdb/default.nix [hunk 3]
```diff
@@ -5,19 +5,14 @@
...
+    sha256 = "0k7fm6gz8m62fkd63k81h9bcz4n6vqx6299fyii6fp2r223p1c5z";
   };
 
-  patches = [
-    (fetchurl {
-        url = "https://github.com/rethinkdb/rethinkdb/commit/871bd3705a1f29c4ab07a096d562a4b06231a97c.patch";
-        sha256 = "05nagixlwnq3x7441fhll5vs70pxppbsciw8qjqp660bdb5m4jm1";
-    })
-  ];
+  patches = [ ./v8-no-snapshot.patch ];
```
Comment (jonringer) [inline pkgs/servers/nosql/rethinkdb/default.nix:15]: do you mind adding a comment about why this patch is necessary?

Comment (matthias-t) [reply pkgs/servers/nosql/rethinkdb/default.nix:16]: Will that suffice? I'm not certain how exactly RethinkDB uses V8 or why the snapshots segfault. When I removed the failing patch, which originally disabled V8 snapshots, I noticed that making V8 snapshots would still segfault the build, so I adapted it for the most recent version of RethinkDB, upon which the build succeeded. But I'm unsure I understand what's going on. The commit message of the original patch is "Workaround for building V8 with GCC 6.2".

Comment (matthias-t) [reply pkgs/servers/nosql/rethinkdb/default.nix:16]: Maybe we could ask @pbogdan? 26ebbac2c42f2d2bb898e9669b2459ff63d0563e

Comment (matthias-t) [reply pkgs/servers/nosql/rethinkdb/default.nix:16]: The associated PR mentions this issue upstream rethinkdb/rethinkdb#5757, which was closed with mandating `CXX=clang++`.

Comment (matthias-t) [reply pkgs/servers/nosql/rethinkdb/default.nix:16]: OK, this builds perfectly using `clangStdenv`, so the latest commit switches to that. Let me know if for some reason you prefer using GCC with the patch.

File: pkgs/servers/nosql/rethinkdb/default.nix [hunk 4]
```diff
@@ -36,6 +31,8 @@ stdenv.mkDerivation rec {
     "--lib-path=${jemalloc}/lib"
   ];
 
+  makeFlags = [ "rethinkdb" ];
```
Comment (matthias-t) [inline pkgs/servers/nosql/rethinkdb/default.nix:34]: > Fix build by building only the database server. Other make targets fetch
dependencies at build time and this behaviour cannot be overriden.
Therefore, the clients and web interface are no longer built. See
rethinkdb/rethinkdb#6867.

Is this ok?

**Comment by Elyhaka** [comment]
Is there anything blocking this PR ? Would be nice to have `rethinkdb` back for the 20.09 ?

**Review by jonringer — COMMENTED** [review]
do you mind squashing the two `rethinkdb` commits?

I see both of them as being a "fix"

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 87828 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>4 packages built:</summary><br>- python27Packages.rethinkdb<br>- python37Packages.rethinkdb<br>- python38Packages.rethinkdb<br>- rethinkdb</details>

**Pull request merged**